GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitCommandsTests.cs

```diff
@@ -45,6 +46,75 @@ public void StatusTest()
             this.ValidateGitCommand("status");
         }
 
+        [TestCase]
+        public void StatusWithGitTraceToPathTest()
+        {
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+            string expectedPath = Path.Combine(this.Enlistment.EnlistmentRoot, "trace-log-expected.txt");
+            string actualPath = Path.Combine(this.Enlistment.EnlistmentRoot, "trace-log-actual.txt");
+
+            string command = "status";
+            string controlRepoRoot = this.ControlGitRepo.RootPath;
+            string gvfsRepoRoot = this.Enlistment.RepoRoot;
+
+            environmentVariables["GIT_TRACE"] = expectedPath;
+            ProcessResult expectedResult = GitProcess.InvokeProcess(controlRepoRoot, command, environmentVariables);
+
+            environmentVariables["GIT_TRACE"] = actualPath;
+            ProcessResult actualResult = GitHelpers.InvokeGitAgainstGVFSRepo(gvfsRepoRoot, command, environmentVariables);
+
+            GitHelpers.ValidateGitOutput(command, expectedResult, actualResult);
+
+            this.FileSystem.FileExists(expectedPath).ShouldBeTrue();
+            this.FileSystem.FileExists(actualPath).ShouldBeTrue();
+
+            string lineToFind = "trace: built-in: git status";
+            this.FileSystem.ReadAllText(expectedResult.Output).IndexOf(lineToFind).ShouldBeAtLeast(0);
+            this.FileSystem.ReadAllText(actualResult.Output).IndexOf(lineToFind).ShouldBeAtLeast(0);
+        }
+
+        [TestCase]
+        public void StatusWithGitTraceToStderrTest()
+        {
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+            environmentVariables["GIT_TRACE"] = "2";
+
+            string command = "status";
+            string controlRepoRoot = this.ControlGitRepo.RootPath;
+            string gvfsRepoRoot = this.Enlistment.RepoRoot;
+
+            ProcessResult expectedResult = GitProcess.InvokeProcess(controlRepoRoot, command, environmentVariables);
+            ProcessResult actualResult = GitHelpers.InvokeGitAgainstGVFSRepo(gvfsRepoRoot, command, environmentVariables);
+
+            GitHelpers.ValidateGitOutput(command, expectedResult, actualResult);
+
+            string.IsNullOrWhiteSpace(expectedResult.Errors).ShouldBeFalse();
+            string.IsNullOrWhiteSpace(actualResult.Errors).ShouldBeTrue();

```

Comment:
Why did one write to Errors but the other didn't?  Shouldn't the `ValidateGitOutput` be checking the Errors?

Reply:
`ValidateGitOutput` does check the errors, which is why this test was failing.

Reply:
These tests are testing the wrong thing :(. The environment stuff we are checking is only for Git commands invoked from GVFS, while these tests are calling Git directly (not going through the code paths I'm checking). That's why they are failing.